ios/Classes/ThumbnailGenerator.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  ThumbnailGenerator.swift
+//  workmanager
+//
+//  Created by Kymer Gryson on 13/08/2019.
+//
+
+import Foundation
+
+struct ThumbnailGenerator {
+    
+    enum ThumbnailIcon {
+        case startWork
+        case success
+        case failure
+        
+        var textValue: String {
+            switch self {
+            case .startWork:
+                return ["👷‍♀️", "👷‍♂️"].randomElement()!
+            case .success:
+                return "🎉"
+            case .failure:
+                return "🔥"
+            }
+        }
+    }
+    
+    @available(iOS 10.0, *)
+    static func createThumbnail(with icon: ThumbnailIcon) -> UNNotificationAttachment? {
+        let name = "thumbnail"
+        let thumbnailFrame = CGRect(x: 0, y: 0, width: 150, height: 150)
+        let thumbnail = UIView(frame: thumbnailFrame)
+        thumbnail.isOpaque = false
+        let label = UILabel(frame: thumbnailFrame)
+        label.text = icon.textValue
+        label.font = UIFont.systemFont(ofSize: 125)
+        label.textAlignment = .center
+        thumbnail.addSubview(label)
+        
+        do {
+            let thumbnailImage = try thumbnail.renderAsImage()
+            let localURL = try thumbnailImage.persist(fileName: name)
```

Comment:
Optimize thumbnail generation: the images are saved to a file because that local file URL is necessary to create the `UNNotificationAttachment`. In subsequent calls the local file URL should be checked first if it still exists, before generating a new thumbnail

Reply:
Discussed with @petermnt: extra complexity not worth the marginal gains.